tensorflow_probability/python/experimental/mcmc/particle_filter_test.py

```diff
@@ -734,6 +735,207 @@ def marginal_log_likelihood(level_scale, noise_scale):
     self.assertAllNotNone(grads)
     self.assertAllAssertsNested(self.assertNotAllZero, grads)
 
+  def test_smc_squared_rejuvenation_parameters(self):
+    def particle_dynamics(params, _, previous_state):
+      reshaped_params = tf.reshape(params,
+                                   [params.shape[0]] +
+                                   [1] * (previous_state.shape.rank - 1))
+      broadcasted_params = tf.broadcast_to(reshaped_params,
+                                           previous_state.shape)
+      return normal.Normal(previous_state + broadcasted_params + 1, 0.1)
+
+    def rejuvenation_criterion(step, state):
+      # Rejuvenation every 2 steps
+      cond = tf.logical_and(
+          tf.equal(tf.math.mod(step, tf.constant(2)), tf.constant(0)),
+          tf.not_equal(state.extra[0], tf.constant(0))
+      )
+      return tf.cond(cond, lambda: tf.constant(True),
+                     lambda: tf.constant(False))
+
+    inner_observations = tf.range(30, dtype=tf.float32)
+
+    num_outer_particles = 3
+    num_inner_particles = 7
+
+    loc = tf.broadcast_to([0., 0.], [num_outer_particles, 2])
+    scale_diag = tf.broadcast_to([0.05, 0.05], [num_outer_particles, 2])
+
+    params, _ = self.evaluate(particle_filter.smc_squared(
+        inner_observations=inner_observations,
+        inner_initial_state_prior=lambda _, params:
+        mvn_diag.MultivariateNormalDiag(
+            loc=loc, scale_diag=scale_diag
+        ),
+        initial_parameter_prior=normal.Normal(3., 1.),
+        num_outer_particles=num_outer_particles,
+        num_inner_particles=num_inner_particles,
+        outer_rejuvenation_criterion_fn=rejuvenation_criterion,
+        inner_transition_fn=lambda params: (
+            lambda _, state: independent.Independent(
+                particle_dynamics(params, _, state), 1)
+        ),
+        inner_observation_fn=lambda params: (
+            lambda _, state: independent.Independent(
+                normal.Normal(state, 2.), 1)
```

Comment:
It seems like the shapes in this test might not all line up.  In particular, it looks like the observation distribution has event_shape [2], but the observation at a single time step is scalar (because observations has shape `[30]`).

It looks like the plan is for the inner particle filter to have a latent space of shape [2], and for both dimensions of the latent space to be observed at a time step (with some noise).  And for there to be a single scalar outer parameter. 
 In that case, I would expect some of the other shapes to look like (given there are 3 outer particles and 7 inner particles):

 - inner particles: shape [3, 7, 2]

 - outer particles/params: shape [3]

 - observations: shape [30, 2]  (for 30 steps)

 - observation_fn: (array[3], array[3, 7, 2]) -> distribution with batch_shape [3, 7] and event_shape [2]

 - inner_transition_fn: (array[3], array[3, 7, 2]) -> distribution with batch shape [3, 7] and event_shape [2]
    * (In the current code, looks like this distribution currently has event_shape [].)

 -  inner_initial_state_prior: array[3] -> distribution with batch shape [3] and event_shape [2] -- and the SMC2 code should take care of drawing enough samples and reshaping to get an initial state of shape [3, 7, 2].
    * (In the current code, it looks like this function returns a distribution with batch shape [7].)


(This should make it possible for this test to pass without the "if particles_dim == 1:" block in `_compute_observation_log_weights`.)